File: 0014_longest_common_prefix/sorting/python/solution.py
from typing import List

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        
        # Lexicographical sort
        strs.sort()
        
        # After sorting, the common prefix of all strings
        # is the common prefix of the first and last strings.
        first = strs[0]
        last = strs[-1]
        
        i = 0
        while i < len(first) and i < len(last) and first[i] == last[i]:
            i += 1
            
        return first[:i]

File: 0014_longest_common_prefix/sorting/cpp/solution.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <cassert>

class Solution {
public:
    std::string longestCommonPrefix(std::vector<std::string>& strs) {
        if (strs.empty()) return "";
        
        std::sort(strs.begin(), strs.end());
        
        std::string first = strs[0];
        std::string last = strs.back();
        
        int i = 0;
        while (i < first.length() && i < last.length() && first[i] == last[i]) {
            i++;
        }
        
        return first.substr(0, i);
    }
};

void test() {
    Solution sol;
    std::vector<std::string> strs1 = {"flower", "flow", "flight"};
    assert(sol.longestCommonPrefix(strs1) == "fl");
    std::cout << "All tests passed!" << std::endl;
}

int main() {
    test();
    return 0;
}
